grpc/server/cmake/external/cares/CMakeLists.txt

```diff
@@ -0,0 +1,18 @@
+cmake_minimum_required(VERSION 2.8)
+
+project(external-cares)
+
+include(ExternalProject)
+
+ExternalProject_add(
+    cares
+    GIT_REPOSITORY https://github.com/c-ares/c-ares.git
```

Comment:
Why is this needed? Doesn't gRPC already come with c-ares?

Reply:
Well, I had to add the options in `grpc/server/cmake/external/grpc/CMakeLists.txt` one by one because it was requesting them for some reason. But honestly I am not completely sure why it could not use the one from the grpc repo =/.

Actually, in the previous version (the current `develop` branch), it was required to install protobuf separately (and to set PROTOBUF_DIR). Not sure why exactly.

Reply:
Well, I am actually checking this, and I think I needed the "PACKAGE" lines in order to "install" grpc. But I cannot reproduce it correctly.

Reply:
So, I figured c-ares has to be compiled as a package because of this message:

> gRPC_INSTALL will be forced to FALSE because gRPC_CARES_PROVIDER is
  "module"

Moreover, if I don't put protobuf as a package, somehow `gRPCTargets.cmake` is not generated.

I want grpc to get installed in `build/default/external/grpc`, so I need them as packages. Maybe there is another way using the in-tree versions and without installing grpc, but in this case I am not sure if the targets would be generated...

The way I see it is that either way, c-ares and grpc are dependencies. Here we have control on the protobuf version we use, and this setup works. If you feel like it is much cleaner to use the in-tree dependencies of grpc, I can try it. But I am not sure it is worth it (especially because the way I do it in this PR is another way, and not a hack).